src/basics/Basics.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace basics;
+
+use InvalidArgumentException;
+
+class Basics implements BasicsInterface
+{
+    private BasicsValidator $validator;
+
+    public function __construct(BasicsValidator $validator)
```

Comment:
Please change it to public function __construct(private BasicsValidator $validator)